spec/board_spec.rb

```diff
@@ -8,52 +8,142 @@
+    it "can validate if the assigned coordinates are on the board" do
+      @board.cells
+      expect(@board.valid_coordinate?("A1")).to eq(true)
+      expect(@board.valid_coordinate?("D4")).to eq(true)
+    end
+
+    it "can validate if un-assigned coordinates are on the board" do
+      @board.cells
+      expect(@board.valid_coordinate?("A5")).to eq(false)
+      expect(@board.valid_coordinate?("E1")).to eq(false)
+      expect(@board.valid_coordinate?("A22")).to eq(false)
+    end
   end
-end
 
-describe "#ValidatingPlacements" do
-  before(:each) do
-    @cruiser = Ship.new("Cruiser", 3)
-    @submarine = Ship.new("Submarine", 2)  
+
+  describe "#ValidatingPlacements" do
+    before(:each) do
+      @cruiser = Ship.new("Cruiser", 3)
+      @submarine = Ship.new("Submarine", 2)  
+    end
+
+    it "checks that the number of coordinates in the array argument are the same length as the ship" do
+      expect(@board.valid_placement?(@cruiser, ["A1", "A2"])).to eq(false)
+      expect(@board.valid_placement?(@submarine, ["A2", "A3", "A4"])).to eq(false)
+    end
+    
+    it "ensures that coordinates are in consecutive order" do
+      expect(@board.valid_placement?(@cruiser, ["A1", "A2", "A4"])).to eq(false)
+      expect(@board.valid_placement?(@submarine, ["A1", "C1"])).to eq(false)
+      expect(@board.valid_placement?(@cruiser, ["A3", "A2", "A1"])).to eq(false)
+      expect(@board.valid_placement?(@submarine, ["C1", "B1"])).to eq(false)
+    end
+    
+    it "checks that coordinates can't be diagonal" do
+      expect(@board.valid_placement?(@cruiser, ["A1", "B2", "B2"])).to eq(false)
+      expect(@board.valid_placement?(@submarine, ["C2", "D3"])).to eq(false)
+    end
+    it "checks coordinates for valid placement" do
+      expect(@board.valid_placement?(@cruiser, ["A1", "A2"])).to eq(true)
+      expect(@board.valid_placement?(@submarine, ["B1", "C1", "D1"])).to eq(true)
+    end
   end
 
-  it "checks that the number of coordinates in the array argument are the same length as the ship" do
-    expect(@board.valid_placement?(@cruiser, ["A1", "A2"])).to eq(false)
-    expect(@board.valid_placement?(@submarine, ["A2", "A3", "A4"])).to eq(false)
+  describe "#RangeClass" do
+    it 'can confirm that it exists' do
+      range = 3..8
+      expect(range.class).to be_a(Range)
+    end
+
+    it 'can turn a Range into an array and return basic array functions' do
+      range = 3..8
+      array = range.to_a
+      expect(array).to eq([3, 4, 5, 6, 7, 8])
+      expect(array.length).to eq(6)
+      expect(array[3]).to eq(6)
+    end
+
+    it 'can apply with String values' do
+      range = "A".."D"
+      array = range.to_a
+      
+      expect(array).to eq(["A", "B", "C", "D"])
+      expect(array.length).to eq(4)
+      expect(array[0]).to eq("A")
+    end
+
+    it 'can return a character with its Ordinal value' do
+      range = "A".."D"
+      array = range.to_a
+      
+      expect(array[0].ord).to eq(65)
```

Comment:
Unique take on testing. Your method is more DRY, and we will go with that approach.